packages/contracts/contracts/libs/LibStateChannelApp.sol

```diff
@@ -0,0 +1,54 @@
+pragma solidity ^0.4.25;
+
+
+/// @title LibStateChannelApp
+/// @author Liam Horne - <liam@l4v.io>
+/// @notice Contains the structures and enums needed for the AppRegistry
+contract LibStateChannelApp {
+
+  // The mode that the App is currently in from POV of the blockchain
+  enum AppStatus {
+    ON,
+    DISPUTE,
+    OFF
+  }
+
+  // An interface to the App used to make function calls to it
+  struct AppInterface {
+    address addr;
+    bytes4 applyAction;
+    bytes4 getTurnTaker;
+    bytes4 isStateTerminal;
+    bytes4 resolve;
+  }
+
+  // A minimal structure that uniquely identifies a single instance of an App
+  struct AppIdentity {
+    address owner;
+    address[] signingKeys;
+    bytes32 appInterfaceHash;
+    bytes32 termsHash;
+    uint256 defaultTimeout;
+  }
+
+  // A minimal structure representing the state of an App from POV of the blockchain
+  struct AppChallenge {
```

Comment:
I think "challenge" is more descriptive because it includes the state of the app, the timeout of the challenge, the current nonce, who submitted the challenge etc. Perhaps the variable should be renamed to `appChallenges` in the mapping